### ISSUE TITLE
Lambda learning rate scheduler

### ISSUE THREAD

**Issue by FlorentF9** [lhoyer/MIC#60]
I think there is a mistake in the lambda learning rate scheduler:

https://github.com/lhoyer/MIC/blob/master/cls/examples/cdan_mcc_sdat.py#L105C5-L108C95
```python
lr_scheduler = LambdaLR(optimizer, lambda x: args.lr *
                            (1. + args.lr_gamma * float(x)) ** (-args.lr_decay))
lr_scheduler_ad = LambdaLR(
    ad_optimizer, lambda x: args.lr * (1. + args.lr_gamma * float(x)) ** (-args.lr_decay))
```

The lambda in LambdaLR is supposed to return the factor to multiply with the base learning rate, so it should not include a multiplication by `args.lr`. This causes the base lr to be multiplied twice (for example, it produces 0.0001 instead of 0.01

**Comment by lhoyer** [comment]
Yes, you are right about this. However, we kept it this way for a fair comparison with the original implementation of SDAT, which has the same bug: 
https://github.com/val-iisc/SDAT/blob/947a10be56ce7482bd10f138807d1cbf8948e335/examples/cdan_mcc_sdat.py#L103
